qiskit_ibm_runtime/qiskit_runtime_service.py

```diff
@@ -712,6 +713,12 @@ def backends(
         # Set fractional gate flag for use when loading properties or refreshing backend.
         for backend in backends:
             backend.options.use_fractional_gates = use_fractional_gates
+        try:
+            backends = deepcopy(backends)
+        except TypeError:
+            # local simulator cannot be deepcopied due to
+            # `TypeError: cannot pickle '_thread.RLock' object`
+            pass
         return filter_backends(backends, filters=filters, **kwargs)
```

Comment:
I think that instead of deep-copying the list, we should probably iterate over it and do individual copies of the backends. Else if a simulator is found, none of the backends get copied. I was thinking something like:

```suggestion
        return_backends = []
        for backend in backends:
          try:
             return_backends.append(deepcopy(backend))
          except TypeError:
              # local simulator cannot be deepcopied due to
              # `TypeError: cannot pickle '_thread.RLock' object`
              return_backends.append(backend)
        return filter_backends(return_backends, filters=filters, **kwargs)
```

There's probably a more efficient implementation for it, but this would be the gist. What do you think?